crates/bifrost/src/providers/replicated_loglet/provider.rs

```diff
@@ -60,22 +66,72 @@ impl LogletProviderFactory for Factory {
 
     async fn create(self: Box<Self>) -> Result<Arc<dyn LogletProvider>, OperationError> {
         metric_definitions::describe_metrics();
-        Ok(Arc::new(ReplicatedLogletProvider))
+        Ok(Arc::new(ReplicatedLogletProvider::new(
+            self.task_center,
+            self.opts,
+            self.metadata,
+            self.metadata_store_client,
+            self.networking,
+        )))
     }
 }
 
-struct ReplicatedLogletProvider;
+struct ReplicatedLogletProvider {
+    active_loglets: DashMap<(LogId, SegmentIndex), Arc<ReplicatedLoglet>>,
+    _task_center: TaskCenter,
+    _opts: BoxedLiveLoad<ReplicatedLogletOptions>,
```

Comment:
Why is this boxed?

Reply:
Because the actual type cannot be named.